testing/testlog/testlog.go

```diff
@@ -17,16 +18,23 @@ type Hook struct {
 	entries []*logrus.Entry
 }
 
-// NewNullLogger Creates a discarding logger and installs the test hook.
-func NewNullLogger() (*logrus.Logger, *Hook) {
+// New sets up a test logger that produces no output. Use the returned hook to
+// observe and make assertions about what was logged.
+func New() (*logrus.Logger, *Hook) {
```

Comment:
I also considered updating this signature to only return a single thing which implements both hook and logger functionality (likely embedding `*logrus.Logger` into `Hook`), so that it's not necessary to explicitly choose whether you want to discard logs or test their output.

That said, I do think it's a bit clearer this way, if less convenient.